### PR TITLE
Add shadow service's api

### PR DESCRIPTION
**Pull request by zouyingjie**
1. Add ``message ShadowService {}`` and it's configuration message: ``ShadowServiceMySQL``, ``ShadowServiceKafka``, ``ShadowServiceRedis``，``ShadowServiceRabbitMQ``, ``ShadowServiceElasticSearch``.

### CODE REVIEW

File: v1alpha1/meshmodel.proto [hunk 1]
```diff
@@ -501,4 +501,83 @@ message URLRule {
...
+    // Password for access Redis server.
+	string password = 3;
+}
+
+// ShadowServiceRabbitMQ is the rabbitmq configuration for shadow service.
+message ShadowServiceRabbitMQ {
+	// Host is the server addresses for RabbitMQ server.
+	repeated string  hosts = 1;
+
+    // Username for access RabbitMQ server.
```
Comment (benja-wu) [inline v1alpha1/meshmodel.proto]: Maybe we can align the lines with a tab as before.

File: v1alpha1/meshmodel.proto [hunk 2]
```diff
@@ -501,4 +501,83 @@ message URLRule {
...
 	// PolicyRef configures which policy this URLRule references to. 
 	string policyRef = 3;
+}
+
+// Shadow Service is a copy of an existing easemesh service. The test traffic can be scheduled to the shadow service for the test.
+message ShadowService {
+	// Name is the mesh service's name.
+	string name = 1;
+
+	// Namespace is kubernetes namespace where the service is deployed.
```
Comment (zhao-kun) [inline v1alpha1/meshmodel.proto]:
```suggestion
	// Namespace is kubernetes namespace where the service is deployed. The shadow service is deployed at the same namespace as the original service by default.
```

File: v1alpha1/meshmodel.proto [hunk 3]
```diff
@@ -501,4 +501,83 @@ message URLRule {
...
+	ShadowServiceRabbitMQ rabbitmq  = 6;
+
+	// ElasticSearch configuration, optional.
+	ShadowServiceElasticSearch elasticsearch  = 7;
+}
+
+
+// ShadowServiceMySQL is the mysql configuration for shadow service.
+message ShadowServiceMySQL {
+	// Hosts is the jdbc address for MySQL databases.
```
Comment (zhao-kun) [inline v1alpha1/meshmodel.proto]: uri better then host

File: v1alpha1/meshmodel.proto (hunk 1 above)
Comment (zouyingjie) [reply v1alpha1/meshmodel.proto]: updated

File: v1alpha1/meshmodel.proto (hunk 3 above)
Comment (zouyingjie) [reply v1alpha1/meshmodel.proto]: Updated

File: v1alpha1/meshmodel.proto (hunk 2 above)
Comment (zouyingjie) [reply v1alpha1/meshmodel.proto]: Updated.

**Review by benja-wu — APPROVED** [review]
LGTM

**Pull request merged**